src/process_management/error.rs

```diff
@@ -0,0 +1,13 @@
+use std::path::Path;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ProcessManagerError<'a> {
+    #[error("Path '{0}' does not exist")]
+    NonExistent(&'a Path),
+    #[error("'{0}' is a relative path")]
+    RelativePath(&'a Path),
+    #[error("'{0}' is not a directory")]
+    NonDirectory(&'a Path)
```

Comment:
This is okay as is, but do you think it would make sense to have "Path" in all errors that have to do with a path problem, just for clarity? Because without the context clues, it's not obvious that the error is talking about a path for `NonExistent` and `NonDirectory`. You could think it's talking about the process manager itself.

Reply:
So do `NonExistentPath` and `NonDirectoryPath`?

That works for me

Reply:
Yeah. Let's do that.